src/polyfem/solver/forms/ElasticForm.hpp

```diff
@@ -96,5 +120,10 @@ namespace polyfem::solver
 		void compute_cached_stiffness();
 
 		Eigen::VectorXd x_prev_;
+
+		mutable std::vector<utils::Tree> quadrature_hierarchy_;
```

Comment:
mutable?

Reply:
Adaptive refinement happens inside the step valid check, which is a const call